### ISSUE TITLE
INTERNAL ERROR while checking portcran

### ISSUE THREAD

**Issue by DragonSA** [python/mypy#2760]
# Issue
Mypy produces an internal error while parsing portcran.py:
```
Traceback (most recent call last):
  File "/usr/local/bin/mypy", line 6, in <module>
    main(__file__)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/main.py", line 40, in main
    res = type_check_only(sources, bin_dir, options)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/main.py", line 81, in type_check_only
    options=options)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/build.py", line 177, in build
    dispatch(sources, manager)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/build.py", line 1323, in dispatch
    process_graph(graph, manager)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/build.py", line 1461, in process_graph
    process_stale_scc(graph, scc)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/build.py", line 1534, in process_stale_scc
    graph[id].semantic_analysis()
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/build.py", line 1288, in semantic_analysis
    self.manager.semantic_analyzer.visit_file(self.tree, self.xpath)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 243, in visit_file
    self.accept(d)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 2306, in accept
    node.accept(self)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/nodes.py", line 660, in accept
    return visitor.visit_class_def(self)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 553, in visit_class_def
    self.analyze_base_classes(defn)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 773, in analyze_base_classes
    base_types.append(self.object_type())
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 842, in object_type
    return self.named_type('__builtins__.object')
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 855, in named_type
    sym = self.lookup_qualified(qualified_name, None)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 2132, in lookup_qualified
    n = self.lookup(parts[0], ctx)  # type: SymbolTableNode
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 2116, in lookup
    self.name_not_defined(name, ctx)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 2277, in name_not_defined
    self.fail(message, ctx)
  File "/home/dbn/.local/lib/python3.5/site-packages/mypy/semanal.py", line 2289, in fail
    self.errors.report(ctx.get_line(), msg, blocker=blocker)
AttributeError: 'NoneType' object has no attribute 'get_line'

*** INTERNAL ERROR ***

/home/dbn/projects/portcran/ports/core/platform.py:10: error: Internal error -- please report a bug at https://github.com/python/mypy/issues

NOTE: you can use "mypy --pdb ..." to drop into the debugger when this happens.
```
The value of msg is `Name '__builtins__' is not defined`

# How to reproduce
```
# git clone https://github.com/DragonSA/portcran.git
# cd portcran
# mypy --py2 portcran.py
```

**Comment by DragonSA** [comment]
Some further information: I installed mypy by:
```
# git clone https://github.com/python/mypy.git
# cd mypy
# python3.5 setup.py install
```
This appears to not install the typeshed libraries.  If they are installed the above error does not occur.

**Comment by elazarg** [comment]
You should use

     git clone --recurse-submodules https://github.com/python/mypy.git

As described in the README. However, the error message should be clearer.

**Comment by elazarg** [comment]
Trying to reproduce, I get a very good error message:

    $ python setup.py install
    Submodule 'typeshed' not initialized.
    Please run:
      git submodule update --init typeshed

**Comment by DragonSA** [comment]
I think the issue was somewhat complicated:
 - I was updating a port for FreeBSD.  Since the pypi package doesn't include the source I switched to using github.  
 - I used the mypy source from github, but not typeshed.
 - I installed a copy of mypy without typeshed (somehow the warning above did not display.  I did get it  when manually installing it).  
 - I then installed mypy from a cloned repository (off the master branch).  
 - Mypy the binary now had two libraries it could choose from, and chose the former (without typeshed)

I'm closing this issue as the installation was "corrupted", thus valid for the assertion (or previously None reference) to be triggered.
